### PR TITLE
feat(gui): run inference in a separate process and temporarily disable button

### PR DESCRIPTION
**Pull request by Lordmau5**
This PR will change the "Infer" button to run in a thread utilizing PySimpleGui's `window.start_thread` method.

While this is going on, the "Infer" button will be disabled until inference is completed (or fails due to an exception)

This will prevent the GUI from freezing up and will allow the user to already prepare the settings for the next inference
___
Added another commit fixing the "Infer" button from disabling a bit too late, potentially making it possible to quickly run it multiple times in a row (e.g. spam clicking)

### CODE REVIEW

**Comment by 34j** [comment]
Wouldn't it be better to throw the whole thing to the pool?

**Comment by 34j** [comment]
I meant that there would be no particular need to disable the button... what do you think?

**Comment by Lordmau5** [comment]
Okay so I moved the function around, maybe it was because I was lacking 2 newlines inbetween, and now it's working properly with the pool.

I've also noticed ThreadPool is faster than ProcessPool.

> I meant that there would be no particular need to disable the button... what do you think?

Well, in theory there isn't, no. However, I think it might be a good visual indicator for the user to know that something is still going on.

If we don't disable the button, and instead just schedule further inferences afterwards, the user has to wait until the first inference is done until it get's to the next one.
So they might accidentally infer the exact same file a 2nd time, resulting in it getting overwritten.

**Comment by 34j** [comment]
I think this change will cause 3 tasks to run at the same time and not work properly. Also, I would like you to use the ProcessPool.

**Comment by Lordmau5** [comment]
> I think this change will cause 3 tasks to run at the same time and not work properly.

Which tasks in particular?
In my testing it always did infer, enable button, and then play audio in that order. Not at the same time.
`max_workers=1` ensures only 1 can run at a time in that pool

**Comment by Lordmau5** [comment]
Also added a proper log call for when it's finished with the inference which will look something lke this:
image

**Comment by Lordmau5** [comment]
> However, apparently now it doesn't want to schedule the `finish_inference` method anymore... Not even the `LOG.info` inside it...
> (Man I hate Python)

Alright, managed to fix that by utilizing the future's `add_done_callback`.
I have no idea why Pebble is deciding to not want to run that method but hey, this works.

File: src/so_vits_svc_fork/gui.py [hunk 1]
```diff
@@ -541,38 +549,49 @@ def apply_preset(name: str) -> None:
...
+                            device="cpu"
+                            if not values["use_gpu"]
+                            else get_optimal_device(),
+                        ),
+                    )
+                    infer_future.add_done_callback(
+                        lambda _future: after_inference(window, input_path)
                     )
                     if values["auto_play"]:
                         pool.schedule(play_audio, args=[output_path])
```
[the remark this answers is not in the document]
Comment (Lordmau5) [reply src/so_vits_svc_fork/gui.py:597]: I was thinking of that and yes, I'll be doing that in another commit in a bit :D

**Comment by 34j** [comment]
Is this now complete? Can it be merged?

**Comment by Lordmau5** [comment]
It should be complete, yes, but I can do one last test once I'm back home in around 2 hours if you'd like me to.

**Pull request merged**